snapcraft/tests/test_yaml.py

```diff
@@ -614,6 +615,77 @@ def test_license_version_without_license_raises_exception(self):
         self.assertEqual(raised.exception.message, expected_message,
                          msg=self.data)
 
+    def test_valid_security_policy_for_apps(self):
+        self.data['apps'] = {
+            'app1': {
+                'command': 'binary',
+                'security-policy': {
+                    'seccomp': 'file.seccomp',
+                    'apparmor': 'file.apparmor',
+                },
+            },
+        }
+
+        snapcraft.yaml._validate_snapcraft_yaml(self.data)
+
+    def test_valid_security_override_for_apps(self):
+        self.data['apps'] = {
+            'app1': {
+                'command': 'binary',
+                'security-override': {
+                    'read-paths': ['path1', 'path2'],
+                    'write-paths': ['path1', 'path2'],
+                    'abstractions': ['abstraction1', 'abstraction2'],
+                    'syscalls': ['open', 'close'],
+                },
+            },
+        }
+
+        snapcraft.yaml._validate_snapcraft_yaml(self.data)
+
+    def test_valid_caps_for_apps(self):
+        self.data['apps'] = {
+            'app1': {
+                'command': 'binary',
+                'caps': ['cap1', 'cap2'],
+            },
+        }
+
+        snapcraft.yaml._validate_snapcraft_yaml(self.data)
+
+    def test_invalid_security_combinations(self):
+        self.data['apps'] = {
+            'app1': {
+                'command': 'binary',
+                'security-override': {
+                    'read-paths': ['path1', 'path2'],
+                    'write-paths': ['path1', 'path2'],
+                    'abstractions': ['abstraction1', 'abstraction2'],
+                    'syscalls': ['open', 'close'],
+                },
+                'caps': ['cap1', 'cap2'],
+                'security-policy': {
+                    'seccomp': 'file.seccomp',
+                    'apparmor': 'file.apparmor',
+                },
+            },
+        }
```

Comment:
Ah, I read too quickly. However, I can break your code above by removing all but one invalid combo and this test would still pass, correct? Should you be explicitly testing each invalid combination rather than throwing them all together?